euphoria-core/src/main/java/cz/seznam/euphoria/core/client/dataset/InputDataset.java

```diff
@@ -57,6 +61,15 @@ public void persist(DataSink<T> sink) {
         "The input dataset is already stored.");
   }
 
+  /**
+   * Input Dataset doesn't have hints
+   * @return empty set
+   */
+  @Override
+  public Set<OutputHint> getHints() {
+    return Collections.emptySet();
+  }
```

Comment:
Another +1 for moving hints to operator, this useless method will disappear. Moreover, the hints are called `OutputHint`, and `Dataset` has no "output". Operator does.